### ISSUE TITLE
Feature request: add support for composer ^2.0 in drupal-l10n ^1.0

### ISSUE THREAD

**Issue by ndf** [drupal-composer/drupal-l10n#25]
We are upgrading all servers in our DTAP-environments to composer 2.0.
But we are worried that unintentionally we might still have some composer 1.0 installs flying around on some of our DTAP environments.

When we upgrade this package to ^2.0 a hard requirement on composer 2.0 is added.

Can we make this package compatible with composer 2.0 _and_ composer 1.0?
That should be done with this package in version ^1.0. That would be semver proof.

Feature request: add support for `composer ^2.0` in `drupal-l10n ^1.0`

**Comment by FlorentTorregrosa** [comment]
> but we are sure if that happens everywhere.

Sorry I don't understand the end of the sentence.

Initially I wanted to provide on one branch, support for the two versions of Composer. But the change is more than just indicating Composer compatibility in composer.json. The class used to download files has changed between Composer 1 and Composer 2. 

On https://github.com/composer/composer/blob/master/UPGRADE-2.0.md:

> If you used RemoteFilesystem you probably should use HttpDownloader instead now

If you can provide a PR showing compatibility with both Composer versions why not.

**Comment by ndf** [comment]
Hi FlorentTorregrosa, thanks for you reply and the great contributions!

The first sentence was indeed very off. I have reformulated that sentence in the original message to:
`We are upgrading all servers in our DTAP-environments to composer 2.0.
But we are worried that unintentionally we might still have some composer 1.0 installs flying around on some of our DTAP environments.`

With your answer I understand why this is quite an undertaking.
The good thing in the current situation:
* This project is compatible with composer 2.0
* And the breaking change is clearly reflected in the version number (which is upped from 1.0 to 2.0) in this project.

I rephrased the issue-title to ```Feature request: add support for composer ^2.0 in drupal-l10n ^1.0```

Honestly I don't think I am able to create a PR for this.
